Source/ChocolateyGui.Common.Windows/Services/PackageArgumentsService.cs

```diff
@@ -0,0 +1,64 @@
+﻿// --------------------------------------------------------------------------------------------------------------------
+// <copyright company="Chocolatey" file="PackageArgumentsService.cs">
+//   Copyright 2017 - Present Chocolatey Software, LLC
+//   Copyright 2014 - 2017 Rob Reynolds, the maintainers of Chocolatey, and RealDimensions Software, LLC
+// </copyright>
+// --
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Security.Cryptography;
+using System.Threading.Tasks;
+using chocolatey;
+
+namespace ChocolateyGui.Common.Windows.Services
+{
+    public class PackageArgumentsService : IPackageArgumentsService
+    {
+        private static readonly byte[] EntropyBytes = Encoding.UTF8.GetBytes("Chocolatey");
+
+        public string Decrypt(string id, string version)
+        {
+            var packageFolder = "{0}.{1}";
+            var packageId = id;
+            var packageVersion = version;
+            var chocolateyPath = @"C:\ProgramData\chocolatey\.chocolatey";
+            var argumentsPath = Path.Combine(chocolateyPath, packageFolder.format_with(id, version));
+            var argumentsFile = Path.Combine(argumentsPath, ".arguments");
+
+            // Get the arguments decrypted in here and return them
+            try
+            {
+                if (File.Exists(argumentsFile))
+                {
+                    string fileName = argumentsFile;
+                    var encryptedBytes = Convert.FromBase64String(File.ReadAllText(fileName));
+                    var decryptedBytes = ProtectedData.Unprotect(encryptedBytes, EntropyBytes, DataProtectionScope.LocalMachine);
+                    string decryptedChocoArgs = Encoding.UTF8.GetString(decryptedBytes);
+                    return decryptedChocoArgs;
+                }
+                else
+                {
+                    throw new FileNotFoundException();
+                }
+            }
+            catch (FileNotFoundException)
+            {
+                return "Chocolatey argument file not found";
+            }
+            catch (Exception ex)
+            {
+                return ex.Message;

```

Comment:
We may have to do some additional work here to better handle this, but I don't have any great ideas for now, so this is fine just now.